Mailbox/DataLoader.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System;
+using Newtonsoft.Json;
+
+namespace Mailbox
+{
+    public class DataLoader : IDisposable
+    {
+        Stream stream;
+        public DataLoader(System.IO.Stream source)
+        {
+            if (source is null) throw new ArgumentNullException();
```

Comment:
The parameter passed to this exception's constructor should be the parameter name
```suggestion
            if (source is null) throw new ArgumentNullException(nameof(source));
```